### PR TITLE
fix: guard the CLI against unsupported Node.js versions

### PR DESCRIPTION
**Pull request by maxnoller**
## Summary

- **Problem:** `engines.node` is `22.x`, but that only emits a non-blocking `EBADENGINE` warning on install. The repo's `engine-strict` `.npmrc` is never published, and consumer npm defaults `engine-strict` to `false`. So a user on Node 18/20 installs "successfully" and then hits a confusing crash later when the CLI uses a Node 22-only API.
- **Why it matters:** First-run experience. The failure surfaces as an opaque stack trace far from the actual cause (wrong Node), not a clear "use Node 22" message.
- **What changed:** Added `src/node-version-guard.ts` and imported it **first** in `src/cli.ts`. ES module dependencies evaluate in import order, so the version check runs before the heavy module graph loads and exits with a clean message on an unsupported major.
- **What did not change:** `engines.node` (still `22.x`, still emits the install-time warning). No new lifecycle/`preinstall` scripts. No behavior change on supported Node 22.

## Change Type

- [x] Bug fix

## Why a runtime guard instead of `preinstall` / `engine-strict`

`engine-strict` can't be forced on consumers (the repo `.npmrc` isn't shipped). A `preinstall` hook would miss the most common real failure modes: `--ignore-scripts` installs and shells that switch to an older Node *after* a successful Node 22 install. A runtime guard fires at the actual point of failure — running the CLI — and covers all of those. This is what most popular CLIs do (npm/yarn/pnpm/Vite/Next).

It's wired as the first import rather than a separate `bin` shim because `cli.ts` gates `main()` on `isDirectExecution()` and has self-spawn logic keyed on `process.argv[1]`; a first-import side effect gives the same "runs before heavy imports" guarantee with a 2-line change and also covers direct dev entry points (`npm run dev` / `gateway` / `tui`).

## Validation

```bash
npx tsc --noEmit          # only pre-existing TS2307s for uninstalled optional deps (camoufox-js, @ngrok/ngrok)
npx @biomejs/biome check src/cli.ts src/node-version-guard.ts   # clean
npx vitest run --project unit tests/node-version-guard.test.ts tests/cli.test.ts tests/cli-verbosity.test.ts   # 165 passing
```

- **New unit test** (`tests/node-version-guard.test.ts`): accept-22 / reject-20 / reject-24.
- **ES module ordering** verified empirically: the first import fully evaluates (and can `process.exit`) before the second import runs.
- **Real Node via Docker** — compiled the actual guard and mirrored `cli.ts`'s import order (guard first, "heavy graph" second):

  | Node | Result | Exit |
  |------|--------|------|
  | 18.20.8 | clean message, heavy graph **never evaluated** | 1 |
  | 20.20.2 | clean message, heavy graph **never evaluated** | 1 |
  | 22 | `HEAVY MODULE GRAPH EVALUATED` → `CLI BODY RAN` | 0 |

  Message on old Node:
  ```
  hybridclaw requires Node.js 22.x, but this process is running 20.20.2.
  Install Node.js 22 (the current LTS) and run the command again.
  If you use a Node version manager (nvm, fnm, volta, asdf), switch to Node 22 first.
  ```

- **Edge cases checked:** existing `cli`-importing tests (153 in `cli.test.ts`) still pass through the new first import on Node 22 (guard is a no-op on the supported major).
- **Skipped checks:** did not run the full `npm run build` (container build); the guard was compiled and exercised in isolation on real Node 18/20/22 instead.

## Risk Notes

- Security-sensitive paths touched? **No.**
- Gateway, audit, approval, or container boundaries touched? **No.**
- Residual gap (called out for honesty): if a sibling module in the import graph did a *named* import of a Node 22-only builtin export, ESM linking would fail before the guard evaluates. The codebase's 22-specific usage is runtime API calls (covered), not missing named exports, so this is not currently hit.

## Docs And Config Impact

- [x] No docs or config impact

> Note: intentionally did **not** add a CHANGELOG entry per maintainer request on this PR.

🤖 AI-assisted: guard implementation, tests, and Docker validation were drafted with Claude Code; author reviewed the diff.

### CODE REVIEW

File: src/node-version-guard.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  const actualMajor = Number.parseInt(
+    String(nodeVersion).split('.')[0] || '0',
+    10,
+  );
+  if (actualMajor === REQUIRED_NODE_MAJOR) {
+    return { ok: true, requiredMajor: REQUIRED_NODE_MAJOR, actualMajor };
+  }
+
+  const message = [
+    `hybridclaw requires Node.js ${REQUIRED_NODE_MAJOR}.x, but this process is running ${nodeVersion}.`,
```
[the remark this answers is not in the document]
Comment (maxnoller) [reply src/node-version-guard.ts]: Good catch — fixed in aa655fa5. `checkNodeVersion` now strips a leading `v` (so both `process.versions.node` and `process.version` work) and falls back to major `0` when the version is unparseable. Added unit tests for the `v22.1.0` and unparseable cases.

[the remark this answers is not in the document]
Comment (maxnoller) [reply src/node-version-guard.ts]: Good catch — fixed in aa655fa5. `checkNodeVersion` now strips a leading `v` (so both `process.versions.node` and `process.version` work) and falls back to major `0` when the version is unparseable. Added unit tests for the `v22.1.0` and unparseable cases.

File: src/node-version-guard.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    `Install Node.js ${REQUIRED_NODE_MAJOR} (the current LTS) and run the command again.`,
+    'If you use a Node version manager (nvm, fnm, volta, asdf), switch to Node 22 first.',
+  ].join('\n');
+
+  return {
+    ok: false,
+    requiredMajor: REQUIRED_NODE_MAJOR,
+    actualMajor,
```
[the remark this answers is not in the document]
Comment (maxnoller) [reply src/node-version-guard.ts]: This was already resolved before this review landed: the guard now sources the required major from package.json `engines.node` and templates every message line with `${requiredMajor}` (the line now reads "switch to Node ${requiredMajor} first"), so there are no hard-coded version literals left.

[the remark this answers is not in the document]
Comment (maxnoller) [reply src/node-version-guard.ts]: This was already resolved before this review landed: the guard now sources the required major from package.json `engines.node` and templates every message line with `${requiredMajor}` (the line now reads "switch to Node ${requiredMajor} first"), so there are no hard-coded version literals left.

**Review by furukama — APPROVED** [review]
